File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Molecules/Dropdown/MolDropdownMenuItem.swift
import Resources
import SwiftUI

public struct MolDropdownMenuItem: View {
    private let text: String
    private let icon: ImageAsset
    private let onTap: () -> Void
    private let isDestructive: Bool

    public init(text: String, icon: ImageAsset, onTap: @escaping () -> Void, isDestructive: Bool = false) {
        self.text = text
        self.icon = icon
        self.onTap = onTap
        self.isDestructive = isDestructive
    }

    public var body: some View {
        Button(role: isDestructive ? .destructive : nil, action: onTap) {
            Label {
                AtomText(text: text, textFontRole: .bodyP1, colorRole: textColorRole)
            } icon: {
                AtomIcon(icon: icon, iconSize: .m, colorRole: iconColorRole)
            }
        }
    }

    private var textColorRole: TextColorRole {
        switch isDestructive {
        case true: .destructive
        case false: .onSurface
        }
    }

    private var iconColorRole: IconColorRole {
        switch isDestructive {
        case true: .destructive
        case false: .onSurface
        }
    }
}

File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/molecules/dropdown/MolDropdownMenuItem.kt
package com.perrystreet.woof.designsystem.atomic.molecules.dropdown

import androidx.annotation.DrawableRes
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.runtime.Composable
import com.perrystreet.woof.designsystem.atomic._tokens.sizing.SizingRoles
import com.perrystreet.woof.designsystem.atomic.atoms.icon.AtomIcon
import com.perrystreet.woof.designsystem.atomic.atoms.icon.roles.IconColorRole
import com.perrystreet.woof.designsystem.atomic.atoms.text.AtomText
import com.perrystreet.woof.designsystem.atomic.atoms.text.roles.TextColorRole
import com.perrystreet.woof.designsystem.atomic.atoms.text.roles.TextFontRole

@Composable
fun MolDropdownMenuItem(
    text: String,
    @DrawableRes iconRes: Int,
    onTap: () -> Unit,
    isDestructive: Boolean = false,
) {
    val textColorRole = when (isDestructive) {
        true -> TextColorRole.Destructive
        false -> TextColorRole.OnSurface
    }
    val iconColorRole = when (isDestructive) {
        true -> IconColorRole.Destructive
        false -> IconColorRole.OnSurface
    }
    DropdownMenuItem(
        text = { AtomText(text = text, textFontRole = TextFontRole.BodyP1, colorRole = textColorRole) },
        leadingIcon = { AtomIcon(iconRes = iconRes, iconSize = SizingRoles.Icon.M, colorRole = iconColorRole) },
        onClick = onTap,
    )
}
